### PR TITLE
PLATFORM-1579: getMenuHelper: properly encode query conditionals

### PR DESCRIPTION
**Pull request by macbre**
@Grunny

### CODE REVIEW

File: includes/wikia/GlobalFunctions.php [hunk 1]
```diff
@@ -619,17 +619,19 @@ function getMenuHelper( $name, $limit = 7 ) {
...
+	$res = $dbr->query( $query, __METHOD__ );
 	$result = array();
 	while ( $row = $dbr->fetchObject( $res ) ) {
 		$result[] = $row->cl_from;
 	}
 	if ( count( $result ) < $limit ) {
-		$query = "SELECT cl_from FROM categorylinks WHERE cl_to = '" . addslashes( $name ) . "' " . ( count( $result ) > 0 ? " AND cl_from NOT IN (" . implode( ',', $result ) . ") " : "" ) . " LIMIT " . ( $limit - count( $result ) );
-		$res = $dbr->query( $query );
+		$resultEscaped = $dbr->makeList( $result ); # PLATFORM-1579 - e.g. 'a', 'b', 'c'
+		$query = "SELECT cl_from FROM categorylinks WHERE cl_to = '" . addslashes( $name ) . "' " . ( count( $result ) > 0 ? " AND cl_from NOT IN (" . $resultEscaped . ") " : "" ) . " LIMIT " . ( $limit - count( $result ) );
```
Comment (Grunny) [inline includes/wikia/GlobalFunctions.php]: `$name` should be using `addQuotes`.

File: includes/wikia/GlobalFunctions.php [hunk 2]
```diff
@@ -619,17 +619,19 @@ function getMenuHelper( $name, $limit = 7 ) {
 	}
 
 	$name = str_replace( " ", "_", $name );
+	$limit = intval( $limit );
 
-	$dbr =& wfGetDB( DB_SLAVE );
+	$dbr = wfGetDB( DB_SLAVE );
 	$query = "SELECT cl_from FROM categorylinks USE INDEX (cl_from), page_visited USE INDEX (page_visited_cnt_inx) WHERE article_id = cl_from AND cl_to = '" . addslashes( $name ) . "' ORDER BY COUNT DESC LIMIT $limit";
```
Comment (Grunny) [inline includes/wikia/GlobalFunctions.php]: `$name` should be using `addQuotes` here too.

**Comment by Grunny** [comment]
Other than the two `addQuotes` uses, LGTM.

File: includes/wikia/GlobalFunctions.php (hunk 1 above)
Comment (macbre) [inline includes/wikia/GlobalFunctions.php]: Used `$dbr->addQuotes`

File: includes/wikia/GlobalFunctions.php (hunk 2 above)
Comment (macbre) [inline includes/wikia/GlobalFunctions.php]: Used `$dbr->addQuotes`

**Comment by macbre** [comment]
Merged into production in 5e8a59e

**Pull request merged**